community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -636,12 +637,15 @@ private TransactionLogModule buildTransactionLogs(
             return -1;
         };
         final LogFileInformation logFileInformation =
-                new PhysicalLogFileInformation( logFiles, logHeaderCache, transactionIdStore::getLastCommittedTransactionId, logInformation );
+                new PhysicalLogFileInformation( logFiles, logHeaderCache,
+                        transactionIdStore::getLastCommittedTransactionId, logInformation );
 
-        String pruningConf = config.get(
-                config.get( GraphDatabaseFacadeFactory.Configuration.ephemeral )
-                ? GraphDatabaseFacadeFactory.Configuration.ephemeral_keep_logical_logs
-                : GraphDatabaseSettings.keep_logical_logs );
+        if ( config.get( GraphDatabaseFacadeFactory.Configuration.ephemeral ) )
+        {
+            config = config.withDefaults(
+                    stringMap( GraphDatabaseSettings.keep_logical_logs.name(), "1 files" ) );
```

Comment:
This piece of code is the logical (and readable) equivalent of the code which it replaced:

```java
        String pruningConf = config.get(
                 config.get( GraphDatabaseFacadeFactory.Configuration.ephemeral )
                 ? GraphDatabaseFacadeFactory.Configuration.ephemeral_keep_logical_logs
                 : GraphDatabaseSettings.keep_logical_logs );
```

That piece of code referenced a piece of duplicated code (which you also commented on) in `GraphDatabaseFacadeFactory` which was:

```java
ephemeral_keep_logical_logs = setting( "dbms.tx_log.rotation.retention_policy", STRING, "1 " + "files", ....)
```

So the code boils down to:

if database is ephemeral
then default for keep_logical_logs is "1 file"
else the regular default defined in GraphDatabaseSettings